cmd/buildah/pull.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/projectatomic/buildah"
+	"github.com/projectatomic/buildah/pkg/parse"
+	util "github.com/projectatomic/buildah/util"
+	"github.com/urfave/cli"
+)
+
+var (
+	pullFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "authfile",
+			Usage: "path of the authentication file. Default is ${XDG_RUNTIME_DIR}/containers/auth.json",
+		},
+		cli.StringFlag{
+			Name:  "cert-dir",
+			Value: "",
+			Usage: "use certificates at the specified path to access the registry",
+		},
+		cli.StringFlag{
+			Name:  "creds",
+			Value: "",
+			Usage: "use `[username[:password]]` for accessing the registry",
+		},
+		cli.BoolFlag{
+			Name:  "quiet, q",
+			Usage: "don't output progress information when pulling images",
+		},
+		cli.StringFlag{
+			Name:  "shm-size",
+			Usage: "size of '/dev/shm'. The format is `<number><unit>`.",
+			Value: "65536k",
+		},
+		cli.StringFlag{
+			Name:  "signature-policy",
+			Usage: "`pathname` of signature policy file (not usually used)",
+		},
+		cli.BoolTFlag{
+			Name:  "tls-verify",
+			Usage: "require HTTPS and verify certificates when accessing the registry",
+		},
+	}
+	pullDescription = "Pulls an image from the specified location"
+
+	pullCommand = cli.Command{
+		Name:           "pull",
+		Usage:          "Pull an image from the specified location",
+		Description:    pullDescription,
+		Flags:          append(pullFlags),
+		Action:         pullCmd,
+		ArgsUsage:      "IMAGE",
+		SkipArgReorder: true,
+	}
+)
+
+func pullCmd(c *cli.Context) error {
+	args := c.Args()
+	if len(args) == 0 {
+		return errors.Errorf("an image name must be specified")
+	}
+	if len(args) > 1 {
+		return errors.Errorf("too many arguments specified")
+	}
+	if err := parse.ValidateFlags(c, pullFlags); err != nil {
+		return err
+	}
+
+	systemContext, err := parse.SystemContextFromOptions(c)
+	if err != nil {
+		return errors.Wrapf(err, "error building system context")
+	}
+
+	pullPolicy := buildah.PullAlways
+
+	signaturePolicy := c.String("signature-policy")
+
+	store, err := getStore(c)
+	if err != nil {
+		return err
+	}
+
+	commonOpts, err := parse.CommonBuildOptions(c)
+	if err != nil {
+		return err
+	}
+
+	transport := buildah.DefaultTransport
+	arr := strings.SplitN(args[0], ":", 2)
+	if len(arr) == 2 {
+		if _, ok := util.Transports[arr[0]]; ok {
+			transport = arr[0]
+		}
+	}
+
+	options := buildah.BuilderOptions{
+		FromImage:             args[0],
+		Transport:             transport,
+		PullPolicy:            pullPolicy,
+		SignaturePolicyPath:   signaturePolicy,
+		SystemContext:         systemContext,
+		DefaultMountsFilePath: c.GlobalString("default-mounts-file"),
+		ImageOnly:             true,
+		CommonBuildOpts:       commonOpts,
```

Comment:
Do these values get used anywhere?

Reply:
yes, eventually in new.go:newBuilder() via the NewBuilder() call at line 115.

Reply:
Oops, I misread, you're asking about commonOpts, not 'options'.  commonOpts don't seem to be used in the image creation, I'll try removing.